io-engine/src/eventing/nexus_events.rs

```diff
@@ -0,0 +1,52 @@
+use events_api::event::{
+    EventAction,
+    EventCategory,
+    EventMessage,
+    EventMeta,
+    EventSource,
+    RebuildStatus,
+};
+
+use crate::{
+    bdev::nexus,
+    eventing::{Event, RebuildEvent},
+    grpc::node_name,
+};
+
+// Nexus event messages from Nexus data.
+impl<'n> Event for nexus::Nexus<'n> {
+    fn event(&self, event_action: EventAction) -> EventMessage {
+        let event_source = EventSource::new(node_name(&None));
```

Comment:
What is `node_name(None)` doing?

Reply:
It gives the hostname from env variable.

Reply:
Getting node_name through MayastorEnvironment::global_or_default().node_name now.